File: app/javascript/src/hooks/get_company_sizes.ts
import { useState, useEffect } from 'react';
import { CompanySize } from '@customtypes/company';

export const useCompanySizes = () => {
  const [companySizes, setCompanySizes] = useState<CompanySize[]>([]);
  const [loading, setLoading] = useState<boolean>(true);
  const [error, setError] = useState<string | null>(null);

  useEffect(() => {
    const fetchCompanySizes = async () => {
      try {
        setLoading(true);
        setError(null);

        const response = await fetch('/company_sizes.json');
        if (!response.ok) {
          throw new Error(`Error fetching company sizes: ${response.status}`);
        }

        const data = await response.json();
        setCompanySizes(data);
      } catch (err) {
        setError(
          err instanceof Error ? err.message : 'An unknown error occurred'
        );
      } finally {
        setLoading(false);
      }
    };

    fetchCompanySizes();
  }, []);

  return { companySizes, loading, error };
};

File: app/python/hooks/get_company_sizes.py
#  app/python/hooks/get_company_sizes.py

import requests


def fetch_company_sizes():
    try:
        response = requests.get("http://localhost:3000/company_sizes.json")
        response.raise_for_status()
        return response.json()
    except requests.exceptions.RequestException as e:
        print(f"Error fetching company sizes: {e}")
        return None


company_size_data = fetch_company_sizes()


def get_company_size(size_range):
    if not company_size_data:
        return None

    for size in company_size_data:
        if size["size_range"] == size_range:
            return size["id"]
    return None
